libdrgn/python/thread.c

```diff
@@ -0,0 +1,91 @@
+#include "drgnpy.h"
+
+static void Thread_dealloc(Thread *self)
+{
+	Py_DECREF(self->prog);
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static PyLongObject *Thread_get_tid(Thread *self)
+{
+	return (PyLongObject *)PyLong_FromLong(self->thread->tid);
+}
+
+static PyByteArrayObject *Thread_get_prstatus(Thread *self)
+{
+	return (PyByteArrayObject *)PyByteArray_FromStringAndSize(
+		self->thread->prstatus.str, self->thread->prstatus.len);
+}
+
+static StackTrace *Thread_stack_trace(Thread *self)
+{
+	struct drgn_error *err;
+	struct drgn_stack_trace *trace;
+	err = drgn_program_stack_trace(&self->prog->prog, self->thread->tid,
+				       &trace);
+	if (err)
+		return set_drgn_error(err);
+	StackTrace *stack_trace =
+		(StackTrace *)StackTrace_type.tp_alloc(&StackTrace_type, 0);
+	stack_trace->trace = trace;
+	return stack_trace;
+}
+
+Thread *create_thread(Program *program, struct drgn_thread *drgn_thread)
+{
+	Thread *thread = (Thread *)Thread_type.tp_alloc(&Thread_type, 0);
+	thread->thread = drgn_thread;
+	Py_INCREF(program);
+	thread->prog = program;
+	return thread;
+}
+
+static PyGetSetDef Thread_getset[] = {
+	{"tid", (getter)Thread_get_tid, NULL, drgn_Thread_tid_DOC},
+	{"prstatus", (getter)Thread_get_prstatus, NULL, drgn_Thread_prstatus_DOC},
+	{},
+};
+
+static PyMethodDef Thread_methods[] = {
+	{"stack_trace", (PyCFunction)Thread_stack_trace, METH_NOARGS,
+	 drgn_Thread_stack_trace_DOC},
+	{},
+};
+
+PyTypeObject Thread_type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	.tp_name = "_drgn.Thread",
+	.tp_basicsize = sizeof(Thread),
+	.tp_dealloc = (destructor)Thread_dealloc,
+	.tp_flags = Py_TPFLAGS_DEFAULT,
+	.tp_getset = Thread_getset,
+	.tp_methods = Thread_methods,
+};
+
+static void ThreadIterator_dealloc(ThreadIterator *self)
+{
+	Py_DECREF(self->prog);
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static Thread *ThreadIterator_next(ThreadIterator *self)
+{
+	struct drgn_thread *drgn_thread;
+	struct drgn_error *err;
+	err = drgn_thread_iterator_next(self->iterator, &drgn_thread);
+	if (err)
+		return set_drgn_error(err);
+	if (drgn_thread == NULL)
+		return NULL;
+	return create_thread(self->prog, drgn_thread);
+}
+
+PyTypeObject ThreadIterator_type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	.tp_name = "_drgn.ThreadIterator",
```

Comment:
This is an implementation detail, so let's give it a leading underscore:

```suggestion
	.tp_name = "_drgn._ThreadIterator",
```